engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -225,19 +228,14 @@ class WorkflowActor(val workflowId: WorkflowId,
   }
 
   when(ExecutingWorkflowState) {
-    case Event(WorkflowExecutionSucceededResponse(jobKeys, outputs),
+    case Event(response @ WorkflowExecutionSucceededResponse(jobKeys, outputs),
     data @ WorkflowActorData(_, Some(workflowDescriptor), _, _)) =>
+      statsActor ! response
```

Comment:
ToL The pattern here and elsewhere seems to be to send the stats actor a notification before the WA has really done the processing for the state, in particular before the finalization.  Is there a danger that the stats actor will think the workflow has finished in a happy place but in reality a failed finalization will fail the workflow?